usr/lib/systemd/system/stas-config@.service

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2022, Dell Inc. or its subsidiaries.  All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+# See the LICENSE file for details.
+#
+# This file is part of NVMe STorage Appliance Services (nvme-stas).
+#
+[Unit]
+Description=nvme-stas /etc/nvme/%i auto-generation
+Documentation=man:stas-config@.service(8)
+ConditionFileNotEmpty=|!/etc/nvme/%i
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/stasadm %i -f /etc/nvme/%i
+
+[Install]
+WantedBy=stas-config.target
```

Comment:
Careful; any systemd units of `Type=oneshot` that are used as dependencies for other units really, really must have `RemainAfterExit=true`.  xref https://github.com/ostreedev/ostree/pull/1697

Reply:
Actually, not in this case. In fact, I followed the example you previously referenced (https://src.fedoraproject.org/rpms/openssh/blob/rawhide/f/sshd-keygen%40.service). In this example, they do not use `RemainAfterExit`.

Reply:
Hmm, right...it may be that the problem only occurs when a `.service` depends on another `.service`, not in the case of a `.target` depending on a `.service`.  Although I'm not totally confident about that.